### PR TITLE
DSPDC-1041 Add mapping for dd_ fields

### PR DESCRIPTION
**Pull request by danxmoran**
Feel free to rip the messy code, took some shortcuts to get it working

### CODE REVIEW

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala [hunk 1]
```diff
@@ -146,6 +172,274 @@ object HLESurveyTransformationPipelineBuilder extends PipelineBuilder[Args] {
...
+        val formYear = rawRecord.getRequired("dd_dog_current_year_calc").toInt
+        val birthYear = rawRecord.getRequired("dd_dog_birth_year").toInt
+
+        val formMonth = rawRecord.getRequired("dd_dog_current_month_calc").toInt
+        val exactMonthKnown = rawRecord.getRequired("dd_dog_birth_month_yn") == "1"
+        val birthMonth = if (exactMonthKnown) {
+          rawRecord.getRequired("dd_dog_birth_month").toInt
+        } else {
+          // ¯\_(ツ)_/¯
+          6
```
Comment (quazi-broad) [inline hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala:218]: lol is this temporary?

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala [hunk 2]
```diff
@@ -146,6 +172,274 @@ object HLESurveyTransformationPipelineBuilder extends PipelineBuilder[Args] {
...
+          ddBreedMixedSecondary = rawRecord.getOptional("dd_dog_breed_mix_2")
+        )
+      }
+    }
+
+    val withAge = {
+      val ageCertain = rawRecord.getRequired("dd_dog_birth_year_certain") == "1"
+
+      /*
+       * FIXME: How do we encode this in the data?
```
Comment (quazi-broad) [inline hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala:200]: I think you did already address this below, when you assign values to ddAgeBasis.

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala (hunk 1 above)
Comment (snwessel) [reply hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala:218]: love it lol

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala (hunk 2 above)
Comment (danxmoran) [reply hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala:200]: Ah, I meant more how we encode the lookup table itself (so people know how to interpret the magic strings). I'll clarify the comment.

File: hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala (hunk 1 above)
Comment (danxmoran) [reply hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HLESurveyTransformationPipelineBuilder.scala:218]: I think it accurately captures my feelings of using `6` here. I'll add some words.

**Pull request merged**